src/main/java/fruitshop/service/handler/PurchaseOperation.java

```
@@ -0,0 +1,18 @@
+package fruitshop.service.handler;
+
+import fruitshop.service.common.FruitTransaction;
+import fruitshop.storage.Storage;
+
+public class PurchaseOperation implements OperationHandler {
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        int currentQuantity = Storage.getFruitStorage().get(transaction.getFruit());
+        if (!(currentQuantity - transaction.getQuantity() < 0)) {
+            Storage.subtract(transaction.getFruit(), transaction.getQuantity());
+        } else {
+            System.out.println("Can not sell this product. "
+                    + "To small quantity in storage");
```

This violates checklist item: 'Handling Purchase operation.' and 'Don't ignore exceptions in try-catch blocks'. If the purchase would result in a negative balance, you must throw a RuntimeException, not just print a message. Also, you should validate that the fruit name is not null or empty.